water_neuron/src/state.rs

```diff
@@ -540,6 +540,39 @@ impl State {
             .is_none());
     }
 
+    pub fn record_neuron_merge(
```

Comment:
This is wrong, you're not removing from wherever it is. Before accepting this request you should make sure that the neuron is not disbursed or else you mint nICP for free. Depending of the state of the withdrawal you should reject it. For example if the neuron is 3 months old, it should be removed from `withdrawal_to_disburse`.